ragaai_catalyst/tracers/agentic_tracing/upload/trace_uploader.py

```
@@ -110,9 +117,57 @@
+                expired_keys.append(cache_key)
+
+        for key in expired_keys:
+            del _dataset_cache[key]
+
+        if expired_keys:
+            logger.info(f"Cleaned up {len(expired_keys)} expired dataset cache entries")
+
 def process_upload(task_id: str, filepath: str, hash_id: str, zip_path: str, 
                   project_name: str, project_id: str, dataset_name: str, 
-                  user_details: Dict[str, Any], base_url: str, timeout=120, fail_on_trace_error=True) -> Dict[str, Any]:
+                  user_details: Dict[str, Any], base_url: str, tracer_type, timeout=120, fail_on_trace_error=True) -> Dict[str, Any]:
```

Flake8 (7.2.0)

[error] continuation line under-indented for visual indent

(E128)